File: segundos 4 lenguajes/js/JugadorDefensa.js
export class JugadorDefensa {
    constructor(nombre = "Defensor", nivel = 1, copia = null) {
        if (copia) {
            this.nombre = copia.nombre;
            this.nivel = copia.nivel;
        } else {
            this.nombre = nombre;
            this.nivel = nivel;
        }
    }

    defender() {
        throw new Error("Método defender() debe implementarse.");
    }
}

File: segundos 4 lenguajes/c#/ProyectoCsharp/JugadorDefensa.cs
namespace ProyectoJugadores
{
    public class JugadorDefensa
    {
        public string Nombre { get; set; }
        public int Nivel { get; set; }

        public JugadorDefensa()
        {
            Nombre = "";
            Nivel = 0;
        }

        public JugadorDefensa(string nombre, int nivel)
        {
            Nombre = nombre;
            Nivel = nivel;
        }

        public static JugadorDefensa Copy(JugadorDefensa j)
        {
            return new JugadorDefensa(j.Nombre, j.Nivel);
        }
    }
}
